src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/product_v2/product.yml

```diff
@@ -65,3 +65,12 @@ admin_products_v2_delete:
             id_product: productId
     requirements:
         productId: \d+
+
+admin_get_ajax_products:
+    path: /search/{languageCode}
+    methods: [GET, POST]
+    defaults:
+        _controller: PrestaShopBundle\Controller\Admin\Sell\Catalog\Product\ProductController::getAjaxProductsAction
+        _legacy_controller: AdminProducts
+    requirements:
+        languageCode: ^[a-zA-Z]{2}(-[a-zA-Z]{2})?$
```

Comment:
Do we need to exclude invalid language codes or should we accept anything, and let the controller returns a 400 response if given language code cannot be found ?

Reply:
I think the sooner it is validated the better, with you can't even generate the URL unless you respect the appropriate format. But I'll add an extra check in the controller since the format alone doesn't ensure that the language is installed

Reply:
OK with that but keep in mind this means we have here a hardcoded validation rule of "what is a valid language code".

if tomorrow we modify what is a valid language code in our business logic, we'll have to also modify it here.

Not 100% DRY 😉

Reply:
You're right I'm gonna modify https://github.com/PrestaShop/PrestaShop/blob/3d06748975e00c548e940bf82862d9d211ae94d4/src/Core/Domain/Language/ValueObject/TagIETF.php a little so that it contains a const that we can use here